contrib/devtools/interactive-test-runner.py

```diff
@@ -0,0 +1,237 @@
+    %(prog)s wallet_ rpc_
+
+All options (--option) that do not target this test runner are passed
+down to the scripts being invoked.
+"""
+
+import argparse
+import os
+import random
+import subprocess
+import sys
+import tempfile
+import time
+
+
+def colorizer(mode):
+    if os.name == 'posix':
+        return lambda s: '\033[' + mode + 'm' + s + '\033[0m'
+    else:
+        return lambda s: s
+
+
+bold = colorizer('1')
+gray = colorizer('1;30')
+red = colorizer('0;31')
+yellow = colorizer('0;33')
+blue = colorizer('0;34')
+
+try:
+    # Make sure python thinks it can write unicode to its stdout
+    '\u2713'.encode('utf8').decode(sys.stdout.encoding)
+    TICK = '✓'
+    CROSS = '✖'
+    CIRCLE = '○'
+except UnicodeDecodeError:
+    TICK = 'P'
+    CROSS = 'x'
+    CIRCLE = 'o'
+
+TEST_EXIT_PASSED = 0
+TEST_EXIT_SKIPPED = 77
+
+
+class RunningJob:
+
+    def __init__(self, test, log_stdout, log_stderr, proc):
+        self.test = test
+        self.log_stdout = log_stdout
+        self.log_stderr = log_stderr
+        self.proc = proc
+        self.started = time.time()
+
+
+class TestResult:
+
+    def __init__(self, test, started, returncode, stdout, stderr):
+        self.test = test
+        self.started = started
+        self.finished = time.time()
+        self.returncode = returncode
+        self.stdout = stdout
+        self.stderr = stderr
+
+    def format(self, test_name_length):
+        if self.returncode == TEST_EXIT_SKIPPED:
+            color = gray
+            status = CIRCLE + ' Skipped'
+        elif self.returncode == 0 and not self.stderr:
+            color = blue
+            status = TICK + ' Success'
+        else:
+            color = red
+            status = CROSS + ' Failed'
+
+        name = self.test.ljust(test_name_length)
+        status = status.ljust(9)
+        duration = self.finished - self.started
+
+        return color, '%s | %s | %.0f s' % (name, status, duration)
+
+
+class InteractiveTestRunner:
+
+    def __init__(self, num_jobs, passon_args, tests_order):
+        assert num_jobs >= 1
+
+        self.repo_root = self.find_git_root()
+        self.functional_test_root = os.path.join(self.repo_root, 'test', 'functional')
+        self.tests = self.read_test_runner()
+        self.num_jobs = num_jobs
+        self.passon_args = passon_args
+        self.line_length = 0
+        self.tests_order = tests_order
+        self.test_name_length = len(max(self.tests, key=len))
+
+    @staticmethod
+    def run_command(*args, **kwargs):
+        proc = subprocess.check_output(args, encoding='utf8', **kwargs)
+        return proc.stdout.strip()
+
+    def find_git_root(self):
+        return self.run_command('git', 'rev-parse', '--show-toplevel')
+
+    def find_functional_tests(self):
+        stdout = self.run_command('git', 'ls-files', '*.py', cwd=self.functional_test_root)
+        pyfiles = filter(lambda line: '/' not in line, stdout.split('\n'))
+        return list(pyfiles)
```

Comment:
you could write instead:
```python
pyfiles = [line for line in stdout.split('\n') if '/' not in line]
```
so then we can avoid the following conversion to `list`.

Reply:
Changed in 3094b9a707c45a2250e0b7128b91000d3609e422